src/Controller/AgendaController.php

```diff
@@ -252,7 +252,38 @@ public function edit(Agenda $agenda, Request $request): ?Response
 
         $form->handleRequest($request);
         if ($form->isSubmitted() && $form->isValid() && !$isFinishedAgenda) {
+            /** @var Agenda $agenda */
             $agenda = $form->getData();
+
+            // Ensure required properties are set if changing to status Ready (Klar).
+            if (AgendaStatus::READY === $agenda->getStatus()) {
+                $startingFlashes = sizeof($request->getSession()->getFlashBag()->keys());
+
+                if (empty($agenda->getDate())) {
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a date as ready', [], 'agenda'));
+                }
+
+                if (empty($agenda->getStart())) {
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a start time as ready', [], 'agenda'));
+                }
+
+                if (empty($agenda->getEnd())) {
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a end time as ready', [], 'agenda'));
+                }
+
+                if (empty($agenda->getAgendaMeetingPoint())) {
+                    $this->addFlash('warning', new TranslatableMessage('Cannot mark agenda without a meeting point as ready', [], 'agenda'));
+                }
+
+                // Check if any flashes has been added
+                $endingFlashes = sizeof($request->getSession()->getFlashBag()->keys());
+                if ($endingFlashes > $startingFlashes) {
```

Comment:
You should not use (the number of) flash messages for control flow – you assume something about how the flash messages are handled internally and you cannot rely on that.

Use a boolean variable, `hasMessages`, say, or collect the messages in a list and check the length of this list, e.g.

```php
$messages = [];

if (empty(…)) {
  $messages['warning'] = new TranslatableMessage(…);
}

if (!empty($messages)) {
  foreach ($messages as $type => $msgs) {
    $this->addFlash($type, $message);
  }
  return $this->redirectToRoute(…);
}
```

Reply:
Updated as proposed.